### ISSUE TITLE
[Feature request]: Add templating functionality for custom response

### ISSUE THREAD

**Issue by qbx2** [frain-dev/convoy#1947]
First of all, thank you for effort to build this great product.

### Is your feature request related to a problem? Please describe.
I want to solve okta's one-time verification request (https://developer.okta.com/docs/concepts/event-hooks/#one-time-verification-request) using convoy, but the challenge requires to respond with header value (x-okta-verification-challenge). 

### Describe the solution you'd like
Add templating functionality just like helm does. It's known to be using golang's template package (https://pkg.go.dev/text/template). It is likely to be easy to integrate and could be familiar to users.

### Describe alternatives you've considered
There is a lot of template engines such as mustache, and I think it has simpler grammar.

### Additional context
Okta requests with GET method for that challenge. Is it possible to support GET methods too?

**Comment by jirevwe** [comment]
Hey @qbx2,

This is an interesting problem. We do something like this when specifying idempotency for incoming projects. 
<img width="772" alt="Screenshot 2024-03-09 at 15 07 16" src="https://github.com/frain-dev/convoy/assets/5263355/4a396c0d-4659-4e11-9477-312e0f1b5038">

Is this good enough? Also, would you be willing to open a PR for this?

**Comment by qbx2** [comment]
@jirevwe Yes, that's correct. I'm sorry, but I'm busy with other work, so it won't be possible to submit a PR for the time being.

**Comment by jirevwe** [comment]
That's fine. We'd open a PR and link it to this issue.

**Comment by qbx2** [comment]
Wow, that's so fast!! I just want to double-check that you didn't forget to support GET method to solve Okta's challenge.

**Comment by IntraMail** [comment]
Hi, I'm planning on using convoy but I'm currently stuck by this (mine is slack validation challenge). Does anything has been made to authorize custom body response yet ? 
Thanks a lot
